File: 第七天打卡.py
"""
Definition of ListNode

class ListNode(object):

    def __init__(self, val, next=None):
        self.val = val
        self.next = next
"""

class Solution:
    """
    @param head: n
    @return: The new head of reversed linked list.
    """
    def reverse(self, head):
        # write your code here
        pre = None
        cur = head
        while cur is not None:
            t = cur.next
            cur.next = pre
            pre = cur
            cur = t
        head = pre
        return head

File: 第七天打卡.cpp
public class Solution {
    public ListNode reverse(ListNode head) {
        if(null == head || null == head.next) 
          return head;
        ListNode p = head, q = head;
        while(p.next != null) 
        {
            head = p.next;
            p.next = head.next;
            head.next = q;
            q = head;
        }
        return head;
    }
}
